src/Components/test/E2ETestMigration/Tests/BindTest.cs

```diff
@@ -0,0 +1,1367 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Threading.Tasks;
+using BasicTestApp;
+using Microsoft.AspNetCore.BrowserTesting;
+using Microsoft.AspNetCore.Components.E2ETest.Infrastructure;
+using Microsoft.AspNetCore.Components.E2ETest.Infrastructure.ServerFixtures;
+using Microsoft.AspNetCore.Testing;
+using PlaywrightSharp;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace Microsoft.AspNetCore.Components.E2ETest.Tests
+{
+    public class BindTest : ServerTestBase<ToggleExecutionModeServerFixture<Program>>
+    {
+        public BindTest(
+            ToggleExecutionModeServerFixture<Program> serverFixture,
+            ITestOutputHelper output)
+            : base(serverFixture, output)
+        {
+        }
+
+        protected override Type TestComponent { get; } = typeof(BindCasesComponent);
+
+        protected override async Task InitializeCoreAsync(TestContext context)
+        {
+            await base.InitializeCoreAsync(context);
+
+            // On WebAssembly, page reloads are expensive so skip if possible
+            //Navigate(ServerPathBase, noReload: _serverFixture.ExecutionMode == ExecutionMode.Client);
+        }
+
+        protected async Task<string> GetInputValue(string selector)
+            => await TestPage.EvalOnSelectorAsync<string>(selector, "e => e.value");
+
+        [Theory]
+        [InlineData(BrowserKind.Chromium)]
+        [InlineData(BrowserKind.Firefox)]
+        [InlineData(BrowserKind.Webkit)]
```

Comment:
Will we be repeating this block at the start of every test? Is there some way it could be made into an external concern?

Reply:
The main thing we need is the permutations where we need a test invocation per browser kind.  We could turn each test into a for loop which does each browser test, but that seems like asking for more trouble.  Totally open to different ways to factor these if you have ideas.  The flow right now is the base class initializes the playwright page using reflection to look at the browser kind, so the tests can just work against TestPage.  

Do you favor something like?

```C#
[Fact]
public TestXyz() {
   foreach (var browser in SupportedTestBrowser) {
      var testPage = await GetTestPage(browser);
      // actually testXyz
   }
}
```